### PR TITLE
Add support for session persistence to the load balancer

### PR DESCRIPTION
**Pull request by mouhsen-ibrahim**
This PR adds support for specifying session persistence options when creating/updating load balancers.

# TODO
- [x] Add tests

### CODE REVIEW

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 1]
```diff
@@ -446,6 +457,18 @@ The example below uses OpenStack to create the network, router, a public IP addr
 							Description: descriptions["target_port"],
 							Required:    true,
 						},
+						"session_persistence": schema.SingleNestedAttribute{
+							Description: descriptions["session_persistence"],
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:461]: this description is missing from the `descriptions` map

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 2]
```diff
@@ -446,6 +457,18 @@ The example below uses OpenStack to create the network, router, a public IP addr
...
+							Optional:    true,
+							Computed:    false,
+							Attributes: map[string]schema.Attribute{
+								"use_source_ip_address": schema.BoolAttribute{
+									Description: descriptions["use_source_ip_address"],
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:466]: this description is also missing

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 3]
```diff
@@ -446,6 +457,18 @@ The example below uses OpenStack to create the network, router, a public IP addr
 							Description: descriptions["target_port"],
 							Required:    true,
 						},
+						"session_persistence": schema.SingleNestedAttribute{
+							Description: descriptions["session_persistence"],
+							Optional:    true,
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:462]: In the LoadBalancer OAS seems like every field is required, just wanted to double check that this field is optional

**Comment by vicentepinto98** [comment]
It would be good if you can also add the `session_persistence` field to the loadbalancer `datasource`

File: stackit/internal/services/loadbalancer/loadbalancer_acc_test.go [hunk 4]
```diff
@@ -97,6 +101,7 @@ func configResources(targetPort string) string {
 		loadBalancerResource["interval_jitter"],
 		loadBalancerResource["timeout"],
 		loadBalancerResource["unhealthy_threshold"],
+		loadBalancerResource["use_source_ip_address"],
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer_acc_test.go:104]: this must go after `target_display_name`

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go (hunk 1 above)
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go:463]: done

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go (hunk 2 above)
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go:468]: done

File: stackit/internal/services/loadbalancer/loadbalancer_acc_test.go (hunk 4 above)
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer_acc_test.go]: done

**Comment by mouhsen-ibrahim** [comment]
> You must also add the `session_persistence` field to the loadbalancer `datasource`, because the `Model` is shared between the `resource` and `datasource`, which means if you don't add it there, terraform will throw a `value conversion error`

Done, thanks

File: stackit/internal/services/loadbalancer/loadbalancer/datasource.go [hunk 5]
```diff
@@ -78,33 +78,35 @@ func (r *loadBalancerDataSource) Configure(ctx context.Context, req datasource.C
...
+		"protocol":              "Protocol is the highest network protocol we understand to load balance.",
+		"target_pool":           "Reference target pool by target pool name.",
+		"name":                  "Load balancer name.",
+		"networks":              "List of networks that listeners and targets reside in.",
+		"network_id":            "Openstack network ID.",
+		"role":                  "The role defines how the load balancer is using the network.",
+		"options":               "Defines any optional functionality you want to have enabled on your load balancer.",
+		"acl":                   "Load Balancer is accessible only from an IP address in this range.",
+		"private_network_only":  "If true, Load Balancer is accessible only via a private network IP address.",
+		"session_persistence":   "Here you can setup various session persistence options, so far only use_source_ip_address is supported.",
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/datasource.go:96]:
```suggestion
		"session_persistence":   "Here you can setup various session persistence options, so far only \"`use_source_ip_address`\" is supported.",
```

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 6]
```diff
@@ -166,33 +177,35 @@ func (r *loadBalancerResource) Configure(ctx context.Context, req resource.Confi
...
+		"protocol":              "Protocol is the highest network protocol we understand to load balance.",
+		"target_pool":           "Reference target pool by target pool name.",
+		"name":                  "Load balancer name.",
+		"networks":              "List of networks that listeners and targets reside in.",
+		"network_id":            "Openstack network ID.",
+		"role":                  "The role defines how the load balancer is using the network.",
+		"options":               "Defines any optional functionality you want to have enabled on your load balancer.",
+		"acl":                   "Load Balancer is accessible only from an IP address in this range.",
+		"private_network_only":  "If true, Load Balancer is accessible only via a private network IP address.",
+		"session_persistence":   "Here you can setup various session persistence options, so far only use_source_ip_address is supported.",
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:195]:
```suggestion
		"session_persistence":   "Here you can setup various session persistence options, so far only \"`use_source_ip_address`\" is supported.",
```

**Review by vicentepinto98 — APPROVED** [review]

**Comment by vicentepinto98** [comment]
Acceptance tests passed successfully :white_check_mark:

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**